upi/azurestack/03_infra.json

```diff
@@ -97,15 +97,77 @@
               "frontendPort" : 6443,
               "backendPort" : 6443,
               "probe" : {
-                "id" : "[concat(variables('masterLoadBalancerID'), '/probes/api-internal-probe')]"
+                "id" : "[concat(variables('masterLoadBalancerID'), '/probes/api-public-probe')]"
+              }
+            }
+          }
+        ],
+        "probes" : [
+          {
+            "name" : "api-public-probe",
+            "properties" : {
+              "protocol" : "Tcp",
+              "port" : 6443,
+              "intervalInSeconds" : 10,
+              "numberOfProbes" : 3
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion" : "2017-10-01",
+      "type" : "Microsoft.Network/loadBalancers",
+      "name" : "[variables('internalLoadBalancerName')]",
+      "location" : "[variables('location')]",
+      "sku": {
+        "name": "[variables('skuName')]"
+      },
+      "properties" : {
+        "frontendIPConfigurations" : [
+          {
+            "name" : "internal-lb-ip",
+            "properties" : {
+              "privateIPAllocationMethod" : "Dynamic",
+              "subnet" : {
+                "id" : "[variables('masterSubnetRef')]"
+              },
+              "privateIPAddressVersion" : "IPv4"
+            }
+          }
+        ],
+        "backendAddressPools" : [
+          {
+            "name" : "internal-lb-backend"
```

Comment:
This line defines the name for the backend address pool, which has the name `internal-lb-backend`, so I don't think switching the name to the `internalLoadBalancerName` makes sense. 

Are you suggesting to define a variable for the `backendAddressPool` name? `internal-lb-backend` is used 3 times in this particular template, so I believe it could make sense. On the other hand, I think concatenating variables can get hard to read.

Reply:
I am comparing it to what we are doing for the public load balancer. For that one, we use the public load balancer name as the name of the backend pool. If I look at Azure IPI, I see that the load balancer backend pool uses the cluster name.

Reply:
Ah, I see. The public load balancer name and backend address pool must match according to this: https://github.com/openshift/installer/commit/7c1a274c74ed047d2afcd69c48c663fabda1ee59 

So in IPI they are both named with the cluster name. I made this change in Azure Stack Hub, but it has not been made in public Azure UPI.

The internal load balancer naming currently matches public Azure UPI.

Perhaps I should change all of this load balancer naming in ASH to be consistent with Azure  IPI. That is 
public lb = infra_id 
private lb = infra_id-internal

WDYT?

Reply:
@staebler I have put these changes in 8202e0d1c717a367cd05751f956b0dc566067bdf